app/controllers/result_filters/location_controller.rb

```diff
@@ -2,21 +2,92 @@ module ResultFilters
   class LocationController < ApplicationController
     include FilterParameters
 
+    NO_OPTION = nil
+    LOCATION_OPTION = "1".freeze
+
     def new; end
 
     def create
-      if(!params[:l])
-        flash[:error] = "Please choose an option"
-        redirect_to location_path(params_without_unsupported_location_params)
+      build_params = strip(filter_params.clone)
+
```

Comment:
This would be a good place for the form object pattern we've been discussing. The form object would be active model-esque and have `errors` and `valid?` so the controller would then end up being something like:

```ruby
form_object = LocationForm.new(params)
all_params = filter_params.merge(form_object.params)

if form_object.valid?
  redirect_to results_path(all_params)
else
  flash[:error] = form_object.errors
  redirect_to location_path(all_params)
end
```

Reply:
Yep good idea. I'll make the change

Reply:
Going to refactor in a separate PR

> Failing validation...
> no input and rad selected to 5 miles
> 
> ![image](https://user-images.githubusercontent.com/470137/73670798-ca8ae280-46a1-11ea-8899-f6b38ca40c85.png)
> 
> issues
> 
> * [ ]  the back link got changed...
> * [x]  my previous selection of 5 miles got lost

New ticket for back link bug has been created